lib/maze/schemas/validator_base.rb

```diff
@@ -32,6 +32,58 @@ def validate
         @success = true
       end
 
+      def element_has_value(path, value)
+        element = Maze::Helper.read_key_path(@body, path)
+        if element.nil? || element != value
+          @success = false
+          @errors << "Element '#{path}' was expected to be '#{value}', was '#{element}'"
+        end
+      end
+
+      def element_exists(path)
+        element = Maze::Helper.read_key_path(@body, path)
+        if element.nil?
+          @success = false
+          @errors << "Element '#{path}' was not found"
+        elsif element.kind_of?(Array) && element.empty?
+          @success = false
+          @errors << "Element '#{path}' was an empty array"
```

Comment:
I _think_ this makes sense - we always consider empty arrays as "missing"?  I can imagine that quirks in various platforms could cause an empty array to be serialised from "nothing".

Reply:
I've removed this, and have set it to allow arrays to be marked as present.  Otherwise I can see this potentially being a tripping point as it's not obvious what's happening.